CMakeLists.txt

```diff
@@ -512,6 +512,38 @@ if(ENABLE_HARDENING)
     if(CMAKE_SYSTEM_NAME STREQUAL "Darwin")
       try_append_linker_flag("-Wl,-fixup_chains" TARGET hardening_interface)
     endif()
+
+    if(HAVE_LIBCPP)
+      # https://libcxx.llvm.org/Hardening.html
+      if(CMAKE_BUILD_TYPE STREQUAL "Debug")
```

Comment:
See [here](https://github.com/bitcoin/bitcoin/pull/31458#issuecomment-2541429701) for a more robust approach to the if(debug) test.

Reply:
Yeah, I considered `core_interface_debug`, but I couldn't fit it into this pattern:

```
if debug then
    add compile-definitions-A to hardening_interface
else
    add compile-definitions-B to hardening_interface
```

because

1. the surrounding code adds stuff to `hardening_interface` and I wanted to do the same, not add to `core_interface_debug` and

2. if I add to `core_interface_debug` then in the non-debug case (the `else` branch), where should I add `compile-definitions-B` to? `core_interface` will be used for both release and debug. I think we don't have a symmetric `core_interface_release` which will only be used in release, like `core_interface_debug` is used only in debug.